barretenberg/cpp/pil/vm2/execution.pil

```diff
@@ -89,7 +92,8 @@ sel {
 };
 
 pol commit sel_bytecode_retrieval_success;
-sel * (sel_bytecode_retrieval_success - (1 - sel_bytecode_retrieval_failure)) = 0;
+// If sel = 0, we want sel_bytecode_retrieval_success = 0. We shouldn't be using it.
```

Comment:
At some point I'll add some more comments explaining the cascading. The tricky part is that we want to keep as many zeroes as possible if you didn't get to that stage.